lib/Test/MockFile.pm

```diff
@@ -1770,9 +1769,31 @@ sub __open (*;$@) {
 
     my $abs_path = _find_file_or_fh( $file, 1 );    # Follow the link.
     confess() if !$abs_path && $mode ne '|-' && $mode ne '-|';
-    confess() if $abs_path eq BROKEN_SYMLINK;
+    if ( $abs_path eq BROKEN_SYMLINK ) {
+        $! = ENOENT;
+        return;
+    }
+
     my $mock_file = _get_file_object($abs_path);
 
+    if ( $mock_file && !$mock_file->is_file ) {
+        if ( $mock_file->exists ) {
+            if ( $mock_file->is_dir ) {
+                $! = EISDIR;
+                return;
+            }
+            elsif ( $mock_file->is_link ) {
+                die("Failed to follow link. We shouldn't have gotten here.");
+            }
+            else {
+                die("Unhandled mocked open for a non-file");
+            }
+        }
+        elsif ( defined $mode && substr( $mode, 0, 1 ) eq '>' ) {
```

Comment:
What about `+>` ?

Reply:
@xsawyerx what about:
```perl
elsif ( defined $mode && $mode =~ tr/>+// ) { # If this is an open for write.
```

Reply:
No objections. I keep forgetting about `tr///`. Would you like me to update it?